### PR TITLE
Enhance test coverage for column selection

### PR DESCRIPTION
**Pull request by jrycw**
Related to #35.

This PR aims to enhance test coverage for `cols_move`, `cols_move_to_start`, `cols_move_to_end`, and `cols_hide` functions in both `Pandas` and `Polars`. 

Meanwhile, given the substantial reliance on column selection within the `fmt_*` functions, should we conduct thorough tests for them using `Polars` selectors, or are we satisfied with the current state?

### CODE REVIEW

File: tests/test_spanners.py [hunk 1]
```diff
@@ -231,3 +197,75 @@ def test_cols_width_fully_set_pct_2():
     assert gt_tbl._boxhead[0].column_width == "10%"
     assert gt_tbl._boxhead[1].column_width == "10%"
     assert gt_tbl._boxhead[2].column_width == "40%"
+
+
+@pytest.mark.parametrize("df_lib, columns", [(pd, "a"), (pl, cs.starts_with("a"))])
```
Comment (machow) [inline tests/test_spanners.py:202]: WDYT of making `df_lib` a fixture, that passes in the dataframe directly? Similar to this fixture in tbl_data tests

https://github.com/posit-dev/great-tables/blob/main/tests/test_tbl_data.py#L19-L25

**Review by machow — CHANGES_REQUESTED** [review]
Thanks for this. These tests are so helpful! I left a question about potentially changing `df_lib` to be a fixture. WDYT?

**Comment by machow** [comment]
> Meanwhile, given the substantial reliance on column selection within the fmt_* functions, should we conduct thorough tests for them using Polars selectors, or are we satisfied with the current state?

It seems okay to lean on the tests of `fmt()`, which the `fmt_*()` functions should end up calling (and handles the column selection).

https://github.com/posit-dev/great-tables/blob/main/tests/test_formats.py#L74

File: tests/test_spanners.py (hunk 1 above)
Comment (jrycw) [reply tests/test_spanners.py:202]: Please correct me if I'm mistaken, but it appears that the current test case will generate two small test cases for each test function. However, if we do:

```python
params_frames = [pytest.param(pd.DataFrame, id="pandas"), pytest.param(pl.DataFrame, id="polars")]


@pytest.fixture(params=params_frames, scope="function")
def df(request):
    return request.param({"a": [1, 2], "b": [3, 4], "c": [5, 6], "d": [7, 8]})


@pytest.mark.parametrize("columns", ["a", cs.starts_with("a")])
def test_cols_move_single_col(df, columns):
    src_gt = GT(df)
    new_gt = cols_move(src_gt, columns=columns, after="b")
    assert [col.var for col in new_gt._boxhead] == ["b", "a", "c", "d"]
```

It seems that we'll have four small test cases for each function, resulting in two types of dataframes with two types of columns. It's evident that when `Pandas` interacts with `Polars` columns, the test will fail.

```python
tests/test_spanners.py::test_cols_move_single_col[pandas-a] PASSED                                                                       [ 25%]
tests/test_spanners.py::test_cols_move_single_col[pandas-columns1] FAILED                                                                [ 50%]
tests/test_spanners.py::test_cols_move_single_col[polars-a] PASSED                                                                       [ 75%]
tests/test_spanners.py::test_cols_move_single_col[polars-columns1] PASSED                                                                [100%]

=========================================================== short test summary info ============================================================
FAILED tests/test_spanners.py::test_cols_move_single_col[pandas-columns1] - NotImplementedError: Unsupported selection expr: col("^(a).*$")
1 failed, 3 passed, 1 warning in 0.96s
```

Comment (machow) [reply tests/test_spanners.py:202]: Ah, I didn't realize this is testing specific combinations, where the suggestion just crosses everything. I seems reasonable to keep it as is!

**Review by machow — APPROVED** [review]

**Comment by machow** [comment]
Thanks so much for making sure things are covered! @rich-iannone do you want to merge?

**Pull request merged**